tools/ts_library_builder/main.js

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
+
+// This modules bootstraps ts-node to allow on the fly transpiling of
+// TypeScript while under NodeJS and performs an interpretation of
+// process.argv to pass to the build_library main function.
```

Comment:
I think it would be nice to replace this with a more general tools/ts_library_builder/README.md

Let me take a stab at describing this - please modify

```markdown

This tool allows us to produce a single TypeScript declaration file that describes the complete Deno runtime, including global variables and the built-in "deno" module. The output of this tool, lib.deno_runtime.d.ts, serves several purposes:
1. It is passed to the typescript compiler js/compiler.ts, so that TypeScript knows what types to expect.
2. It is outputted to stdout by `deno --types`, so that users can easily have access to the complete declaration file. Editors can use this in the future to perform type checking.
3. Because JSDocs are maintained, this serves as a simple documentation page for Deno. We will use this file to generate HTML docs in the future.

## Design
Ideally we wouldn't have to build this tool at all, and could simply use tsc to output this declaration file. However, we were not able to get it to output a clean declaration using `emitDeclarationOnly` and `outFile`. 
[.... maybe add more detail here.]
```